src/legacy/core_plugins/ui_metric/public/hacks/ui_metric_init.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// @ts-ignore
+import { uiModules } from 'ui/modules';
+import chrome from 'ui/chrome';
+import { createAnalyticsReporter, setTelemetryReporter } from '../services/telemetry_analytics';
+
+function telemetryInit($injector: any) {
+  const localStorage = $injector.get('localStorage');
```

Comment:
I will test it out, we have this same implementation in other places as well so if this is the case then we might have a bigger issue. Note that I only use storage to load the report from a previous session, i use the in memory object during and keep them in sync all the time, this way if the storage is failing for some reason, only loading previous unreported session report will fail.

Reply:
@bmcconaghy kibana fails to run with cookies/localstorage privacy on safari. But this issue should be tackled in another PR. My code should not change since the fix should be on the localstorage service level which I am using.